File: app/src/main/java/com/example/android/pingpongscorekeeper/adapters/CategoryAdapter.java
package com.example.android.pingpongscorekeeper.adapters;


import android.content.Context;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;

import com.example.android.pingpongscorekeeper.activities.FinishedMatchesFragment;
import com.example.android.pingpongscorekeeper.fragments.NewGameFragment;

public class CategoryAdapter extends FragmentPagerAdapter {

    private Context mContext;

    public CategoryAdapter(Context context, FragmentManager fm) {
        super(fm);
        mContext = context;
    }

    @Override
    public Fragment getItem(int position) {

        if(position == 0) return new NewGameFragment();
        else return new FinishedMatchesFragment();
    }

    @Override
    public int getCount() {
        return 2;
    }

    @Override
    public CharSequence getPageTitle(int position) {
        if(position == 0) return "New Game";
        else return "Finished";
    }
}

File: app/src/main/java/com/example/android/pingpongscorekeeper/adapters/CategoryAdapter.kt
package com.example.android.pingpongscorekeeper.adapters


import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentPagerAdapter

import com.example.android.pingpongscorekeeper.fragments.FinishedMatchesFragment
import com.example.android.pingpongscorekeeper.fragments.NewGameFragment
import com.example.android.pingpongscorekeeper.fragments.PlayersListFragment

class CategoryAdapter(fm: FragmentManager) : FragmentPagerAdapter(fm) {

    override fun getItem(position: Int): Fragment {
        return when (position) {
            0 -> NewGameFragment()
            1 -> FinishedMatchesFragment()
            2 ->  PlayersListFragment()
            else -> throw IllegalStateException("Unknown position $position")
        }
    }

    override fun getCount(): Int  = 3

    override fun getPageTitle(position: Int): String {
        return when (position) {
            0 -> "New Game"
            1 -> "Finished"
            2 -> "Players"
            else -> throw IllegalStateException("Unknown position $position")
        }
    }
}